lib/models/asyncapi.js

```diff
@@ -369,4 +347,50 @@ function assignIdToAnonymousSchemas(doc) {
   schemaDocument(doc, callback);
 }
 
+/**
+ * Recursively go through schema of object type and execute callback.
+ * 
+ * @param {Schema} schema Object type.
+ * @param {Function} callback(schema)
+ *         the function that is called foreach schema found.
+ *         schema {Schema}: the found schema.
+ */
+function recursiveSchemaObject(schema, callback) {
+  if (schema.additionalProperties() !== undefined && typeof schema.additionalProperties() !== 'boolean') {
+    const additionalSchema = schema.additionalProperties();
+    recursiveSchema(additionalSchema, callback);
+  }
+  if (schema.properties() !== null) {
+    const props = schema.properties();
+    for (const [, propertySchema] of Object.entries(props)) {
+      recursiveSchema(propertySchema, callback);
+    }
+  }
+}
+
+/**
+ * Recursively go through schema of array type and execute callback.
+ * 
+ * @param {Schema} schema Array type.
+ * @param {Function} callback(schema)
+ *         the function that is called foreach schema found.
+ *         schema {Schema}: the found schema.
+ */
+function recursiveSchemaArray(schema, callback) {
```

Comment:
Identically as above comment about save common value (here is a `schema.additionalItems()` and `schema.items()`) in const.

Reply:
I did not change the logic when not needed on purpose, I just wanted to get rid of cognitive complexity